### PR TITLE
Document `loaders.py`

### PR DESCRIPTION
**Pull request by bobturneruk**
This is an attempt to make sure `loaders.py` has a complete set of docstrings and that they are in a format usable by sphinx and readthedocs (if and when we implement this).

### CODE REVIEW

File: simple_network_sim/loaders.py [hunk 1]
```diff
@@ -200,40 +263,45 @@ def __getitem__(self, age):
...
                                for key, val in self._entries.items()) + "]"
 
     def __iter__(self):
-        """
+        """Iterate by row.
+
         Iterator that iterates over the matrix keys (a key points to a row). The values returned by the iterator will
         all be strings, since that's how the public interface when indexing the matrix.
         """
         return iter(str(age_range) for age_range in self._entries)
```
[the remark this answers is not in the document]
Comment (bobturneruk) [reply simple_network_sim/loaders.py]: Resolved by a1e9cf2

**Comment by bobturneruk** [comment]
Happy to have feedback on general level of documentedness (too much?) - I feel like I might have overdone the `_` and `__` bits at they're essentially private. Are the extra bits for sphinx distracting? Maybe also I'm exposing myself as not a pure Python person!

**Comment by bobturneruk** [comment]
Sorry about the commit history - got into a gituation.

**Comment by bobturneruk** [comment]
This is ready for another look when time allows, please @WPettersson or @aflag. If you feel my git mess-up should be corrected before merging, let me know and I'll make another branch.

File: simple_network_sim/loaders.py [hunk 2]
```diff
@@ -118,19 +187,23 @@ def _check_overlap(one, two):
 
 
 class AgeRange:
-    """A helper class for an age range."""
+    """A helper class for an age range.
+    
+    If b is None, it is assumed that a is a string to be
+    parsed.
```
Comment (aflag) [inline simple_network_sim/loaders.py:193]: There's also the option of a being a tuple. Although, the feature is not currently used.

File: simple_network_sim/loaders.py [hunk 3]
```diff
@@ -154,41 +227,59 @@ def __init__(self, a, b=None):
...
     def __hash__(self):
+        """Return a hash of the current AgeRange instance."""
         return hash((self._lower, self._upper))
 
 
 class MixingRow:
-    """A mixing row is one row of a mixing table. This is a helper class. A row
+    """One row of a mixing table.
+    
+    This is a helper class. A row
```
Comment (aflag) [inline simple_network_sim/loaders.py:257]: even though it's more change, I think we should probably reformat these lines anyway

File: simple_network_sim/loaders.py [hunk 4]
```diff
@@ -154,41 +227,59 @@ def __init__(self, a, b=None):
...
     represents a given population, and can return the expected number of
     interactions (per day) a member of this population will have with some
```
Comment (aflag) [inline simple_network_sim/loaders.py:259]: I think we could remove the `(per day)` here. That's data-specific there's nothing in the code guaranteeing that.

File: simple_network_sim/loaders.py [hunk 5]
```diff
@@ -86,12 +109,29 @@ def readNodeAttributesJSON(filename):
 # in future - e.g. sampling, generating movements, whatever
 # it should return a networkx graph, ideally with weighted edges
 # eventual replacement with HDF5 reading code?
-def genGraphFromContactFile(filename):
-    G = nx.read_edgelist(filename, create_using=nx.DiGraph, delimiter=",", data=(('weight', float),))
+def genGraphFromContactFile(filename: str) -> nx.DiGraph:
+    """Read a file containing edge weights between nodes.
+
+    Pairs of nodes are listed in the file by source, destination and weight.
```
Comment (aflag) [inline simple_network_sim/loaders.py:115]: sorce, destination, weight and adjustment.

File: simple_network_sim/loaders.py (hunk 2 above)
Comment (bobturneruk) [reply simple_network_sim/loaders.py:193]: Resolved by 6bb334f908d3966e1bedfd1279012636bf8ae132

File: simple_network_sim/loaders.py (hunk 3 above)
Comment (bobturneruk) [reply simple_network_sim/loaders.py:257]: Resolved by 382b7c5867da2ad3ef5fefc5d3f9bf15296f1b86. I think.

File: simple_network_sim/loaders.py (hunk 4 above)
Comment (bobturneruk) [reply simple_network_sim/loaders.py:259]: Changed to "per unit time" f413377d493cd9c7c93f6bc0e479513b6435c93a This OK?

File: simple_network_sim/loaders.py (hunk 5 above)
Comment (bobturneruk) [reply simple_network_sim/loaders.py:115]: Fixed with c51b8957f0a6bc1649b643e9ee296c75172b4296

**Comment by bobturneruk** [comment]
Do not merge. Use #41.

**Pull request closed without merging**